src/persist-client/src/read.rs

```diff
@@ -503,6 +518,16 @@ where
         Ok(Listen::new(self, as_of).await)
     }
 
+    /// Returns a [`BatchFetcher`], which is only capable of fetching batches.
+    ///
+    /// If, after generating a `BatchFetcher`, you have no further need for the
+    /// [`ReadHandle`] from which it's derived, you can
+    /// [expire][`ReadHandle::expire`] it.
```

Comment:
I think it's easy to forget to expire it. Maybe this should always consume the `ReadHandle` and expire it. This will mean that a `Listen` has to clone the handle, to be able to do this. But that might be worth the additional robustness.

Reply:
Ah, I misunderstood the comment about cloning the `ReadHandle` being problematic earlier. Understand now, and will adjust.

Reply:
Null problemo! :blush: